diff --git a/CMakeLists.txt b/CMakeLists.txt
index b787631..1fc8cf4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -402,6 +402,7 @@ add_custom_command (
   COMMAND ${CMAKE_COMMAND} -E touch ${_glog_EMPTY_SOURCE}
 )
 
+target_compile_options(glog_internal PRIVATE "-ffile-prefix-map==.")
 add_library (glog
   $<TARGET_OBJECTS:glog_internal>
   ${_glog_EMPTY_SOURCE}
diff --git a/src/glog/platform.h b/src/glog/platform.h
index 4aaf5b0..f118ca8 100644
--- a/src/glog/platform.h
+++ b/src/glog/platform.h
@@ -42,6 +42,9 @@
 #  ifndef GLOG_OS_LINUX
 #    define GLOG_OS_LINUX
 #  endif
+#  if defined(__ANDROID__)
+#    define GLOG_OS_ANDROID
+#  endif
 #elif defined(macintosh) || defined(__APPLE__) || defined(__APPLE_CC__)
 #  define GLOG_OS_MACOSX
 #elif defined(__FreeBSD__)
@@ -52,9 +55,6 @@
 #  define GLOG_OS_OPENBSD
 #elif defined(__EMSCRIPTEN__)
 #  define GLOG_OS_EMSCRIPTEN
-#elif defined(__ANDROID__)
-#  define GLOG_OS_ANDROID
-#else
 // TODO(hamaji): Add other platforms.
 #error Platform not supported by glog. Please consider to contribute platform information by submitting a pull request on Github.
 #endif
