diff --git a/CMakeLists.txt b/CMakeLists.txt
index 93da0e62..4a535a3d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -209,24 +209,26 @@ if(WITH_RUST)
     list(APPEND ALL_INC ${INC_DIR}/chewing_rs.h)
 endif()
 
-add_subdirectory(doc)
-add_subdirectory(tests)
-add_subdirectory(data)
-
 # library
 add_library(common OBJECT
     src/porting_layer/src/asprintf.h
 
     src/porting_layer/src/asprintf.c
 )
+target_compile_options(common PRIVATE
+    "-ffile-prefix-map=${CMAKE_SOURCE_DIR}=."
+)
 target_compile_definitions(common PRIVATE
-    CHEWING_DATADIR=\"${CMAKE_INSTALL_FULL_DATADIR}/libchewing\"
+    CHEWING_DATADIR=\"./libchewing\"
 )
 
 add_library(libchewing ${ALL_INC} src/chewing.c)
 set_target_properties(libchewing PROPERTIES LINKER_LANGUAGE C)
+target_compile_options(libchewing PRIVATE
+    "-ffile-prefix-map=${CMAKE_SOURCE_DIR}=."
+)
 target_compile_definitions(libchewing PRIVATE
-    CHEWING_DATADIR=\"${CMAKE_INSTALL_FULL_DATADIR}/libchewing\"
+    CHEWING_DATADIR=\"./libchewing\"
 )
 if(NOT WITH_RUST)
     target_sources(common PRIVATE
@@ -279,6 +281,9 @@ if(NOT WITH_RUST)
             ${SRC_DIR}/userphrase-hash.c
         )
     endif()
+    target_compile_options(userphrase PRIVATE
+        "-ffile-prefix-map=${CMAKE_SOURCE_DIR}=."
+    )
     target_link_libraries(libchewing
         PRIVATE common
         PRIVATE userphrase)
