diff --git a/CMakeLists.txt b/CMakeLists.txt
index afac588..7d11d94 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,7 +52,6 @@ add_subdirectory(data/hanja)
 if(ENABLE_EXTERNAL_KEYBOARDS)
 add_subdirectory(data/keyboards)
 endif()
-add_subdirectory(tools)
 add_subdirectory(po)
 
 if(BUILD_TESTING)
diff --git a/hangul/CMakeLists.txt b/hangul/CMakeLists.txt
index cfda59b..d3a82b9 100644
--- a/hangul/CMakeLists.txt
+++ b/hangul/CMakeLists.txt
@@ -45,9 +45,9 @@ add_library(hangul
 
 target_compile_definitions(hangul
     PRIVATE -DHAVE_CONFIG_H
-    PRIVATE -DLOCALEDIR=\"${CMAKE_INSTALL_FULL_LOCALEDIR}\"
-    PRIVATE -DLIBHANGUL_DEFAULT_HANJA_DIC=\"${CMAKE_INSTALL_FULL_DATADIR}/${CMAKE_PROJECT_NAME}/hanja/hanja.txt\"
-    PRIVATE -DLIBHANGUL_DATA_DIR=\"${CMAKE_INSTALL_FULL_DATADIR}/${CMAKE_PROJECT_NAME}\"
+    PRIVATE -DLOCALEDIR=\"${CMAKE_INSTALL_LOCALEDIR}\"
+    PRIVATE -DLIBHANGUL_DEFAULT_HANJA_DIC=\"${CMAKE_INSTALL_DATADIR}/${CMAKE_PROJECT_NAME}/hanja/hanja.txt\"
+    PRIVATE -DLIBHANGUL_DATA_DIR=\"${CMAKE_INSTALL_DATADIR}/${CMAKE_PROJECT_NAME}\"
     PRIVATE -DTOP_SRCDIR=\"${CMAKE_SOURCE_DIR}\"
 )
 
