diff --git a/CMakeLists.txt b/CMakeLists.txt
index 07703bf..5ab6d3b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,7 @@ cmake_minimum_required(VERSION 3.10)
 aux_source_directory(src predict_src)
 
 add_library(rime-predict-objs OBJECT ${predict_src})
+target_compile_options(rime-predict-objs PRIVATE "-ffile-prefix-map=${CMAKE_SOURCE_DIR}=.")
 if(BUILD_SHARED_LIBS)
   set_target_properties(rime-predict-objs
     PROPERTIES
