diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 8faa61f8..671622ce 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -166,6 +166,8 @@ if(BUILD_SHARED_LIBS)
 else()
   add_library(rime-static STATIC ${rime_src})
   target_link_libraries(rime-static ${rime_deps})
+  target_compile_options(rime-static PRIVATE "-ffile-prefix-map=${CMAKE_SOURCE_DIR}=.")
+  target_compile_definitions(rime-static PRIVATE BOOST_DISABLE_CURRENT_LOCATION BOOST_ALL_NO_EMBEDDED_GDB_SCRIPTS)
   set_target_properties(rime-static PROPERTIES
     OUTPUT_NAME "rime" PREFIX "lib"
     ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
diff --git a/src/rime/engine.cc b/src/rime/engine.cc
index 4a54f64c..2efe0c93 100644
--- a/src/rime/engine.cc
+++ b/src/rime/engine.cc
@@ -52,7 +52,6 @@ class ConcreteEngine : public Engine {
   vector<of<Filter>> filters_;
   vector<of<Formatter>> formatters_;
   vector<of<Processor>> post_processors_;
-  an<Switcher> switcher_;
 };
 
 // implementations
diff --git a/src/rime/engine.h b/src/rime/engine.h
index 6b51a68e..c5b8f7e4 100644
--- a/src/rime/engine.h
+++ b/src/rime/engine.h
@@ -16,6 +16,7 @@ namespace rime {
 class KeyEvent;
 class Schema;
 class Context;
+class Switcher;
 
 class Engine : public Messenger {
  public:
@@ -35,6 +36,7 @@ class Engine : public Messenger {
   void set_active_engine(Engine* engine = nullptr) { active_engine_ = engine; }
 
   RIME_DLL static Engine* Create();
+  an<Switcher> switcher_;
 
  protected:
   Engine();
diff --git a/src/rime/lever/deployment_tasks.cc b/src/rime/lever/deployment_tasks.cc
index 6e6ff895..fe111d55 100644
--- a/src/rime/lever/deployment_tasks.cc
+++ b/src/rime/lever/deployment_tasks.cc
@@ -429,17 +429,6 @@ static bool ConfigNeedsUpdate(Config* config) {
 }
 
 bool ConfigFileUpdate::Run(Deployer* deployer) {
-  const path shared_data_path(deployer->shared_data_dir);
-  const path user_data_path(deployer->user_data_dir);
-  // trash deprecated user copy created by an older version of Rime
-  path source_config_path(shared_data_path / file_name_);
-  path dest_config_path(user_data_path / file_name_);
-  path trash = user_data_path / "trash";
-  if (TrashDeprecatedUserCopy(source_config_path, dest_config_path,
-                              version_key_, trash)) {
-    LOG(WARNING) << "deprecated user copy of '" << file_name_
-                 << "' is moved to " << trash;
-  }
   // build the config file if needs update
   the<Config> config(Config::Require("config")->Create(file_name_));
   if (ConfigNeedsUpdate(config.get())) {
diff --git a/src/rime/service.h b/src/rime/service.h
index a69ae741..9b3d4b16 100644
--- a/src/rime/service.h
+++ b/src/rime/service.h
@@ -44,11 +44,11 @@ class Session {
   Schema* schema() const;
   time_t last_active_time() const { return last_active_time_; }
   const string& commit_text() const { return commit_text_; }
-
- private:
-  void OnCommit(const string& commit_text);
-
   the<Engine> engine_;
+
+ private:
+  void OnCommit(const string& commit_text);
+
   time_t last_active_time_ = 0;
   string commit_text_;
 };
diff --git a/src/rime_api_impl.h b/src/rime_api_impl.h
index 30f4514f..c17d4794 100644
--- a/src/rime_api_impl.h
+++ b/src/rime_api_impl.h
@@ -17,6 +17,8 @@
 #include <rime/setup.h>
 #include <rime/signature.h>
 #include <rime/switches.h>
+#include <rime/engine.h>
+#include <rime/switcher.h>
 
 using namespace rime;
 
@@ -430,6 +432,13 @@ RIME_DEPRECATED void RimeSetOption(RimeSessionId session_id,
   if (!ctx)
     return;
   ctx->set_option(option, !!value);
+  auto engine = session->engine_.get();
+  auto switcher = engine->switcher_;
+  if (switcher->IsAutoSave(option)) {
+    if (Config* user_config = switcher->user_config()) {
+      user_config->SetBool("var/option/" + std::string(option), value);
+    }
+  }
 }
 
 RIME_DEPRECATED Bool RimeGetOption(RimeSessionId session_id,
