diff --git a/data/CMakeLists.txt b/data/CMakeLists.txt
index 3b7057aa..f3eb802a 100644
--- a/data/CMakeLists.txt
+++ b/data/CMakeLists.txt
@@ -130,7 +130,9 @@ foreach(DICT ${DICTS})
     FILES
       ${DICT_GENERATED_DIR}/${DICT}.ocd2
     DESTINATION
-      ${DIR_SHARE_OPENCC}
+      data
+    COMPONENT
+      data
   )
 
   set_directory_properties(
@@ -161,5 +163,7 @@ install(
   FILES
     ${CONFIG_FILES}
   DESTINATION
-    ${DIR_SHARE_OPENCC}
+    data
+  COMPONENT
+    data
 )
